python/py_vapid/jwt.py

```diff
@@ -0,0 +1,88 @@
+import binascii
+import json
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import ec, utils
+from cryptography.hazmat.primitives import hashes
+
+from py_vapid.utils import b64urldecode, b64urlencode
+
+
+def extract_signature(auth):
+    """Fix the JWT auth token
+    
+    convert a ecdsa integer pair into an OpenSSL DER pair.
+    
+    :param auth: A JWT Authorization Token.
+    :type auth: str
+    
+    :return tuple containing the signature material and signature
+    
+    """
+    payload, asig = auth.encode('utf8').rsplit(b'.', 1)
+    sig = b64urldecode(asig)
+    if len(sig) != 64:
+        return payload, sig
+
+    encoded = utils.encode_dss_signature(
```

Comment:
Oh, you were telling me about this! Just to make sure I understand, this is to decode OpenSSL signatures, and the `len(sig) != 64` branch above is to decode ones signed with this library?